In [42]:
import os
import pandas as pd
from sklearn.datasets import load_iris
train = pd.read_csv('train.csv') 
#엑셀 다운로드

In [43]:
train.describe()

tBodyAcc-mean()-X  tBodyAcc-mean()-Y  tBodyAcc-mean()-Z  \
count        7352.000000        7352.000000        7352.000000   
mean            0.274488          -0.017695          -0.109141   
std             0.070261           0.040811           0.056635   
min            -1.000000          -1.000000          -1.000000   
25%             0.262975          -0.024863          -0.120993   
50%             0.277193          -0.017219          -0.108676   
75%             0.288461          -0.010783          -0.097794   
max             1.000000           1.000000           1.000000   

       tBodyAcc-std()-X  tBodyAcc-std()-Y  tBodyAcc-std()-Z  tBodyAcc-mad()-X  \
count       7352.000000       7352.000000       7352.000000       7352.000000   
mean          -0.605438         -0.510938         -0.604754         -0.630512   
std            0.448734          0.502645          0.418687          0.424073   
min           -1.000000         -0.999873         -1.000000         -1.000000   
25%           -0.992754         -0.978129         -0.980233         -0.993591   
50%           -0.946196         -0.851897         -0.859365         -0.950709   
75%           -0.242813         -0.034231         -0.262415         -0.292680   
max            1.000000          0.916238          1.000000          1.000000   

       tBodyAcc-mad()-Y  tBodyAcc-mad()-Z  tBodyAcc-max()-X  ...  \
count       7352.000000       7352.000000       7352.000000  ...   
mean          -0.526907         -0.606150         -0.468604  ...   
std            0.485942          0.414122          0.544547  ...   
min           -1.000000         -1.000000         -1.000000  ...   
25%           -0.978162         -0.980251         -0.936219  ...   
50%           -0.857328         -0.857143         -0.881637  ...   
75%           -0.066701         -0.265671         -0.017129  ...   
max            0.967664          1.000000          1.000000  ...   

       fBodyBodyGyroJerkMag-skewness()  fBodyBodyGyroJerkMag-kurtosis()  \
count                      7352.000000                      7352.000000   
mean                         -0.307009                        -0.625294   
std                           0.321011                         0.307584   
min                          -0.995357                        -0.999765   
25%                          -0.542602                        -0.845573   
50%                          -0.343685                        -0.711692   
75%                          -0.126979                        -0.503878   
max                           0.989538                         0.956845   

       angle(tBodyAccMean,gravity)  angle(tBodyAccJerkMean),gravityMean)  \
count                  7352.000000                           7352.000000   
mean                      0.008684                              0.002186   
std                       0.336787                              0.448306   
min                      -0.976580                             -1.000000   
25%                      -0.121527                             -0.289549   
50%                       0.009509                              0.008943   
75%                       0.150865                              0.292861   
max                       1.000000                              1.000000   

       angle(tBodyGyroMean,gravityMean)  angle(tBodyGyroJerkMean,gravityMean)  \
count                       7352.000000                           7352.000000   
mean                           0.008726                             -0.005981   
std                            0.608303                              0.477975   
min                           -1.000000                             -1.000000   
25%                           -0.482273                             -0.376341   
50%                            0.008735                             -0.000368   
75%                            0.506187                              0.359368   
max                            0.998702                     

In [44]:
from sklearn.preprocessing import LabelEncoder
from sklearn.preprocessing import StandardScaler
import numpy as np

label_encoder = LabelEncoder()
train_data = train.drop("Activity", axis=1)
train_data = train_data.drop("subject", axis=1)
train_target = label_encoder.fit_transform(train["Activity"].copy())

scaler = StandardScaler()
train_data_scaled = scaler.fit_transform(train_data.astype(np.float32))



# 테스트 데이터
# test_data = test.drop("Activity", axis=1)
# test_data = test_data.drop("subject", axis=1)
# test_target = label_encoder.transform(test["Activity"].copy())

In [45]:
train_data.describe()

tBodyAcc-mean()-X  tBodyAcc-mean()-Y  tBodyAcc-mean()-Z  \
count        7352.000000        7352.000000        7352.000000   
mean            0.274488          -0.017695          -0.109141   
std             0.070261           0.040811           0.056635   
min            -1.000000          -1.000000          -1.000000   
25%             0.262975          -0.024863          -0.120993   
50%             0.277193          -0.017219          -0.108676   
75%             0.288461          -0.010783          -0.097794   
max             1.000000           1.000000           1.000000   

       tBodyAcc-std()-X  tBodyAcc-std()-Y  tBodyAcc-std()-Z  tBodyAcc-mad()-X  \
count       7352.000000       7352.000000       7352.000000       7352.000000   
mean          -0.605438         -0.510938         -0.604754         -0.630512   
std            0.448734          0.502645          0.418687          0.424073   
min           -1.000000         -0.999873         -1.000000         -1.000000   
25%           -0.992754         -0.978129         -0.980233         -0.993591   
50%           -0.946196         -0.851897         -0.859365         -0.950709   
75%           -0.242813         -0.034231         -0.262415         -0.292680   
max            1.000000          0.916238          1.000000          1.000000   

       tBodyAcc-mad()-Y  tBodyAcc-mad()-Z  tBodyAcc-max()-X  ...  \
count       7352.000000       7352.000000       7352.000000  ...   
mean          -0.526907         -0.606150         -0.468604  ...   
std            0.485942          0.414122          0.544547  ...   
min           -1.000000         -1.000000         -1.000000  ...   
25%           -0.978162         -0.980251         -0.936219  ...   
50%           -0.857328         -0.857143         -0.881637  ...   
75%           -0.066701         -0.265671         -0.017129  ...   
max            0.967664          1.000000          1.000000  ...   

       fBodyBodyGyroJerkMag-meanFreq()  fBodyBodyGyroJerkMag-skewness()  \
count                      7352.000000                      7352.000000   
mean                          0.125293                        -0.307009   
std                           0.250994                         0.321011   
min                          -1.000000                        -0.995357   
25%                          -0.023692                        -0.542602   
50%                           0.134000                        -0.343685   
75%                           0.289096                        -0.126979   
max                           0.946700                         0.989538   

       fBodyBodyGyroJerkMag-kurtosis()  angle(tBodyAccMean,gravity)  \
count                      7352.000000                  7352.000000   
mean                         -0.625294                     0.008684   
std                           0.307584                     0.336787   
min                          -0.999765                    -0.976580   
25%                          -0.845573                    -0.121527   
50%                          -0.711692                     0.009509   
75%                          -0.503878                     0.150865   
max                           0.956845                     1.000000   

       angle(tBodyAccJerkMean),gravityMean)  angle(tBodyGyroMean,gravityMean)  \
count                           7352.000000                       7352.000000   
mean                               0.002186                          0.008726   
std                                0.448306                          0.608303   
min                               -1.000000                         -1.000000   
25%                               -0.289549                         -0.482273   
50%                                0.008943                          0.008735   
75%                                0.292861                          0.506187   
max                                1.000000                          0.998702   

       angle(tBodyGyroJ

In [46]:
corr_matrix = train.corr()
train_data = train.drop("Activity", axis=1) # drop labels for training set
train_data = train_data.drop("subject", axis=1) # 필요없어 보여서 삭제
train_target= train["Activity"].copy()

/var/folders/76/nhbbpw0168g4pxc5wlf6ns2m0000gn/T/ipykernel_6816/3056656915.py:1: FutureWarning: The default value of numeric_only in DataFrame.corr is deprecated. In a future version, it will default to False. Select only valid columns or specify the value of numeric_only to silence this warning.
  corr_matrix = train.corr()


## AdaBoostClassifier

In [47]:
from sklearn.ensemble import AdaBoostClassifier
from sklearn.model_selection import cross_val_score

# Create an instance of AdaBoostClassifier with desired parameters
adaboost_clf = AdaBoostClassifier(
    algorithm='SAMME.R', base_estimator=None,
    learning_rate=1.0, n_estimators=50,
    random_state=123
)

# Fit the model on the training data
adaboost_clf.fit(train_data.values, train_target)

# Use cross_val_score with the new model
cross_val_scores = cross_val_score(adaboost_clf, train_data.values, train_target, cv=10, scoring="accuracy")
print(f"{cross_val_scores.mean()}")


/Users/taewonyoon/anaconda3/lib/python3.11/site-packages/sklearn/ensemble/_base.py:156: FutureWarning: `base_estimator` was renamed to `estimator` in version 1.2 and will be removed in 1.4.
  warnings.warn(
/Users/taewonyoon/anaconda3/lib/python3.11/site-packages/sklearn/ensemble/_base.py:156: FutureWarning: `base_estimator` was renamed to `estimator` in version 1.2 and will be removed in 1.4.
  warnings.warn(
/Users/taewonyoon/anaconda3/lib/python3.11/site-packages/sklearn/ensemble/_base.py:156: FutureWarning: `base_estimator` was renamed to `estimator` in version 1.2 and will be removed in 1.4.
  warnings.warn(
/Users/taewonyoon/anaconda3/lib/python3.11/site-packages/sklearn/ensemble/_base.py:156: FutureWarning: `base_estimator` was renamed to `estimator` in version 1.2 and will be removed in 1.4.
  warnings.warn(
/Users/taewonyoon/anaconda3/lib/python3.11/site-packages/sklearn/ensemble/_base.py:156: FutureWarning: `base_estimator` was renamed to `estimator` in version 1.2 and will b

0.5447493345164153


## DecisionTreeClassifier

In [48]:
from sklearn.tree import DecisionTreeClassifier
from sklearn.model_selection import cross_val_score

# Create an instance of DecisionTreeClassifier with desired parameters
dt_clf = DecisionTreeClassifier(
    ccp_alpha=0.0, class_weight=None, criterion='gini',
    max_depth=None, max_features=None, max_leaf_nodes=None,
    min_impurity_decrease=0.0, min_samples_leaf=1,
    min_samples_split=2, min_weight_fraction_leaf=0.0,
    random_state=123, splitter='best'
)

# Fit the model on the training data
dt_clf.fit(train_data.values, train_target)

# Use cross_val_score with the new model
cross_val_scores = cross_val_score(dt_clf, train_data.values, train_target, cv=10, scoring="accuracy")
print(f"{cross_val_scores.mean()}")


0.8588187666370896


## RandomForestClassifier

In [49]:
from sklearn.ensemble import RandomForestClassifier
from sklearn.model_selection import cross_val_score

# Create an instance of RandomForestClassifier with desired parameters
rf_clf = RandomForestClassifier(
    bootstrap=True, ccp_alpha=0.0, class_weight=None,
    criterion='gini', max_depth=None, max_features='sqrt',
    max_leaf_nodes=None, max_samples=None,
    min_impurity_decrease=0.0, min_samples_leaf=1,
    min_samples_split=2, min_weight_fraction_leaf=0.0,
    n_estimators=100, n_jobs=-1, oob_score=False,
    random_state=123, verbose=0, warm_start=False
)

# Fit the model on the training data
rf_clf.fit(train_data.values, train_target)

# Use cross_val_score with the new model
cross_val_scores = cross_val_score(rf_clf, train_data.values, train_target, cv=10, scoring="accuracy")
print(f"{cross_val_scores.mean()}")


0.931997744750074


## SGDClassifier

In [50]:
from sklearn.linear_model import SGDClassifier
from sklearn.model_selection import cross_val_score

# Create an instance of SGDClassifier with desired parameters
sgd_clf = SGDClassifier(
    alpha=0.0001, average=False, class_weight=None,
    early_stopping=False, epsilon=0.1, eta0=0.001, fit_intercept=True,
    l1_ratio=0.15, learning_rate='optimal', loss='hinge',
    max_iter=1000, n_iter_no_change=5, n_jobs=-1, penalty='l2',
    power_t=0.5, random_state=123, shuffle=True, tol=0.001,
    validation_fraction=0.1, verbose=0, warm_start=False
)

# Fit the model on the training data
sgd_clf.fit(train_data.values, train_target)

# Use cross_val_score with the new model
cross_val_scores = cross_val_score(sgd_clf, train_data.values, train_target, cv=10, scoring="accuracy")
print(f"{cross_val_scores.mean()}")


0.9341709183673469


## LogisticRegression

In [51]:
from sklearn.linear_model import LogisticRegression
from sklearn.model_selection import cross_val_score

# Create an instance of LogisticRegression with desired parameters
logistic_reg_clf = LogisticRegression(
    C=1.0, class_weight=None, dual=False, fit_intercept=True,
    intercept_scaling=1, l1_ratio=None, max_iter=1000,
    multi_class='auto', n_jobs=None, penalty='l2',
    random_state=123, solver='lbfgs', tol=0.0001, verbose=0,
    warm_start=False
)

# Fit the model on the training data
logistic_reg_clf.fit(train_data.values, train_target)

# Use cross_val_score with the new model
cross_val_scores = cross_val_score(logistic_reg_clf, train_data.values, train_target, cv=10, scoring="accuracy")
print(f"{cross_val_scores.mean()}")


/Users/taewonyoon/anaconda3/lib/python3.11/site-packages/sklearn/linear_model/_logistic.py:460: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  n_iter_i = _check_optimize_result(
/Users/taewonyoon/anaconda3/lib/python3.11/site-packages/sklearn/linear_model/_logistic.py:460: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#l

0.9477754362614611


/Users/taewonyoon/anaconda3/lib/python3.11/site-packages/sklearn/linear_model/_logistic.py:460: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  n_iter_i = _check_optimize_result(


In [52]:
from sklearn.ensemble import ExtraTreesClassifier
from sklearn.model_selection import cross_val_score

# ExtraTreesClassifier의 매개변수 설정
extra_trees_params = {
    'bootstrap': False, 'ccp_alpha': 0.0, 'class_weight': None,
    'criterion': 'gini', 'max_depth': None, 'max_features': 'sqrt',
    'max_leaf_nodes': None, 'max_samples': None,
    'min_impurity_decrease': 0.0, 'min_samples_leaf': 1,
    'min_samples_split': 2, 'min_weight_fraction_leaf': 0.0,
    'n_estimators': 100, 'n_jobs': -1, 'oob_score': False,
    'random_state': 123, 'verbose': 0, 'warm_start': False
}

# ExtraTreesClassifier 생성
extra_trees_classifier = ExtraTreesClassifier(**extra_trees_params)

# 모델을 훈련 데이터에 맞춤
extra_trees_classifier.fit(train_data.values, train_target)

# cross_val_score를 사용하여 모델 성능 평가
cross_val_scores = cross_val_score(extra_trees_classifier, train_data.values, train_target, cv=10, scoring="accuracy")
print(f"{cross_val_scores.mean()}")


0.9355351597160604


#### 좋았던 모델들로 앙상블

##### 데이터만 섞어도 성능 향상

In [53]:
from sklearn.ensemble import ExtraTreesClassifier
from sklearn.ensemble import BaggingClassifier
from sklearn.ensemble import VotingClassifier, RandomForestClassifier
from sklearn.linear_model import LogisticRegression
from sklearn.svm import SVC
from sklearn.model_selection import cross_val_score
import numpy as np

# 기본 시드 설정
seed = 1

# 데이터 섞기
np.random.seed(seed)
shuffled_indices = np.random.permutation(len(train_data_scaled))
shuffled_train_data = train_data_scaled[shuffled_indices]
shuffled_indices_train_target = train_target[shuffled_indices]

# 모델 초기화
logistic_regression_classifier = LogisticRegression(solver="liblinear", random_state=0)
bagging_classifier = BaggingClassifier(
    ExtraTreesClassifier(random_state=0, n_jobs=-1), bootstrap=False, random_state=42)
svm_classifier = SVC(gamma='auto', probability=True, random_state=0, kernel="linear")

voting_classifier = VotingClassifier(
    estimators=[('lr', logistic_regression_classifier), ('rf', bagging_classifier), ('svc', svm_classifier)],
    voting='soft'
)

# 교차 검증 수행
cross_val_scores = cross_val_score(voting_classifier, shuffled_train_data, shuffled_indices_train_target, cv=10, scoring='accuracy', n_jobs=-1)
print(f"교차 검증 점수: {cross_val_scores}")
print(f"교차 검증 평균: {np.mean(cross_val_scores)}")


교차 검증 점수: [0.99184783 0.98913043 0.9877551  0.99183673 0.98639456 0.98911565
 0.99183673 0.98367347 0.9877551  0.98503401]
교차 검증 평균: 0.9884379621413784


In [55]:
from sklearn.ensemble import ExtraTreesClassifier
from sklearn.ensemble import BaggingClassifier
from sklearn.ensemble import VotingClassifier, RandomForestClassifier
from sklearn.linear_model import LogisticRegression
from sklearn.svm import SVC
from sklearn.model_selection import cross_val_score
import numpy as np

# 기존 시드 변경
seed = 21401

# 데이터 섞기
np.random.seed(seed)
shuffled_indices = np.random.permutation(len(train_data))
shuffled_train_data = train_data.values[shuffled_indices]
shuffled_indices_train_target = train_target[shuffled_indices]

# 모델 초기화
logistic_regression_classifier = LogisticRegression(solver="liblinear", random_state=0)
bagging_classifier = BaggingClassifier(
    ExtraTreesClassifier(random_state=0, n_jobs=-1), bootstrap=False, random_state=42)
svm_classifier = SVC(gamma='auto', probability=True, random_state=0, kernel="linear")

voting_classifier = VotingClassifier(
    estimators=[('lr', logistic_regression_classifier), ('rf', bagging_classifier), ('svc', svm_classifier)],
    voting='soft'
)

# 교차 검증 수행
cross_val_scores = cross_val_score(voting_classifier, shuffled_train_data, shuffled_indices_train_target, cv=10, scoring='accuracy', n_jobs=-1)
print(f"교차 검증 점수: {cross_val_scores}")
print(f"교차 검증 평균: {np.mean(cross_val_scores)}")


교차 검증 점수: [0.98777174 0.98097826 0.99047619 0.99047619 0.99319728 0.99183673
 0.99591837 0.98639456 0.99047619 0.99319728]
교차 검증 평균: 0.9900722789115646


#### 시드 추가

In [56]:
from sklearn.ensemble import BaggingClassifier

# 기존 시드 변경
seed = 21401

# 데이터 섞기
np.random.seed(seed)
shuffled_indices = np.random.permutation(len(train_data))
shuffled_train_data = train_data.values[shuffled_indices]
shuffled_indices_train_target = train_target[shuffled_indices]

log_clf = LogisticRegression(solver="liblinear", random_state=0)
bag_clf = BaggingClassifier(
    RandomForestClassifier(random_state=0, n_jobs=-1), bootstrap=False, random_state=42)
svm_clf = SVC(gamma='auto', probability=True, random_state=0, kernel="linear")

voting_clf = VotingClassifier(
    estimators=[('lr', log_clf), ('rf', bag_clf), ('svc', svm_clf)],
    voting='soft'
)

    # 교차 검증 수행
cross_val_scores = cross_val_score(voting_clf, shuffled_train_data, shuffled_indices_train_target, cv=10, scoring='accuracy', n_jobs=-1)
print(f"Cross Validation Scores: {cross_val_scores}")
print(f"Cross Validation Mean is {np.mean(cross_val_scores)}")

Cross Validation Scores: [0.99048913 0.98369565 0.99047619 0.99047619 0.99455782 0.99047619
 0.99591837 0.99047619 0.98911565 0.99319728]
Cross Validation Mean is 0.9908878660159715


### 배깅안에 랜포

In [57]:
from sklearn.ensemble import BaggingClassifier

# 기본 시드 설정
seed = 21401

# 데이터 섞기
np.random.seed(seed)
shuffled_indices = np.random.permutation(len(train_data))
shuffled_train_data = train_data.values[shuffled_indices]
shuffled_indices_train_target = train_target[shuffled_indices]

# 모델 초기화
log_clf = LogisticRegression(solver="liblinear", random_state=0)
rf_clf = RandomForestClassifier(n_estimators=115, max_depth=15, random_state=0, n_jobs=-1)
bag_clf = BaggingClassifier(
    RandomForestClassifier(random_state=0, n_jobs=-1), bootstrap=False, random_state=42)
svm_clf = SVC(gamma='auto', probability=True, random_state=0, kernel="linear")

voting_clf = VotingClassifier(
    estimators=[('lr', log_clf), ('rf', bag_clf), ('svc', svm_clf)],
    voting='soft'
)

    # 교차 검증 수행
cross_val_scores = cross_val_score(voting_clf, shuffled_train_data, shuffled_indices_train_target, cv=10, scoring='accuracy', n_jobs=-1)
print(f"Cross Validation Scores: {cross_val_scores}")
print(f"Cross Validation Mean is {np.mean(cross_val_scores)}")

Cross Validation Scores: [0.99048913 0.98369565 0.99047619 0.99047619 0.99455782 0.99047619
 0.99591837 0.99047619 0.98911565 0.99319728]
Cross Validation Mean is 0.9908878660159715


In [ ]:
# from sklearn.ensemble import BaggingClassifier
# from sklearn.ensemble import RandomForestClassifier
# from sklearn.linear_model import LogisticRegression
# from sklearn.svm import SVC
# from sklearn.ensemble import VotingClassifier
# from sklearn.model_selection import cross_val_score
# import numpy as np

# # 기본 시드 설정
# seed = 21401

# # 데이터 섞기
# np.random.seed(seed)
# shuffled_indices = np.random.permutation(len(train_data))
# shuffled_train_data = train_data.iloc[shuffled_indices]
# shuffled_indices_train_target = train_target.iloc[shuffled_indices]

# for i in range(25, 32):
#     # 모델 초기화
#     logistic_regression_classifier = LogisticRegression(solver="liblinear", random_state=12300)
#     random_forest_classifier = RandomForestClassifier(bootstrap=False, n_estimators=120, max_depth=i, random_state=9636, n_jobs=-1)
#     svm_classifier = SVC(C=1, gamma='auto', probability=True, random_state=246, kernel="linear")

#     # Bagging Classifier를 초기화하고 base_estimator에 RandomForestClassifier를 지정
#     bagging_classifier = BaggingClassifier(
#         base_estimator=random_forest_classifier, bootstrap=True, random_state=42, n_jobs=-1
#     )

#     voting_classifier = VotingClassifier(
#         estimators=[('lr', logistic_regression_classifier), ('rf', bagging_classifier), ('svc', svm_classifier)],
#         voting='soft'
#     )

#     # 교차 검증 수행
#     cross_val_scores = cross_val_score(voting_classifier, shuffled_train_data, shuffled_indices_train_target, cv=10, scoring='accuracy', n_jobs=-1)
#     print(f"For max_depth={i}")
#     print(f"Cross Validation Scores: {cross_val_scores}")
#     print(f"Cross Validation Mean is {np.mean(cross_val_scores)}")
#     print()


Cross Validation Scores: [0.98641304 0.98233696 0.99047619 0.99047619 0.99183673 0.99047619
 0.99455782 0.98367347 0.99183673 0.99183673]
Cross Validation Mean is 0.9893920068027212
Cross Validation Scores: [0.98641304 0.98233696 0.99047619 0.99047619 0.99183673 0.99047619
 0.99455782 0.98367347 0.99183673 0.99183673]
Cross Validation Mean is 0.9893920068027212
Cross Validation Scores: [0.98641304 0.98233696 0.99047619 0.99047619 0.99183673 0.99047619
 0.99455782 0.98367347 0.99183673 0.99183673]
Cross Validation Mean is 0.9893920068027212
Cross Validation Scores: [0.98641304 0.98233696 0.99047619 0.99047619 0.99183673 0.99047619
 0.99455782 0.98367347 0.99183673 0.99183673]
Cross Validation Mean is 0.9893920068027212
Cross Validation Scores: [0.98641304 0.98233696 0.99047619 0.99047619 0.99183673 0.99047619
 0.99455782 0.98367347 0.99183673 0.99183673]
Cross Validation Mean is 0.9893920068027212
Cross Validation Scores: [0.98641304 0.98233696 0.99047619 0.99047619 0.99183673 0.9904761

In [58]:
# 시드 설정하기
seed = 21401

# 데이터 섞기
np.random.seed(seed)
shuffled_indices = np.random.permutation(len(train_data))
shuffled_train_data = train_data.values[shuffled_indices]
shuffled_train_target = train_target[shuffled_indices]

# 모델 초기화
logistic_regression_classifier = LogisticRegression(solver="liblinear", random_state=12300)
random_forest_classifier = RandomForestClassifier(bootstrap=False, n_estimators=120, max_depth=30, random_state=9636, n_jobs=-1)
svm_classifier = SVC(C=1, gamma='auto', probability=True, random_state=246, kernel="linear")

voting_classifier = VotingClassifier(
    estimators=[('lr', logistic_regression_classifier), ('rf', random_forest_classifier), ('svc', svm_classifier)],
    voting='soft'
)

# 교차 검증 수행
cross_val_scores = cross_val_score(voting_classifier, shuffled_train_data, shuffled_train_target, cv=10, scoring='accuracy', n_jobs=-1)
print(f"Cross Validation Scores: {cross_val_scores}")
print(f"Cross Validation Mean is {np.mean(cross_val_scores)}")


Cross Validation Scores: [0.99048913 0.98505435 0.99183673 0.99183673 0.99455782 0.99183673
 0.99591837 0.99047619 0.99047619 0.99319728]
Cross Validation Mean is 0.9915679532682639


In [59]:
# 시드 설정하기
seed = 20000

# 데이터 섞기
np.random.seed(seed)
shuffled_indices = np.random.permutation(len(train_data))
shuffled_train_data = train_data.values[shuffled_indices]
shuffled_train_target = train_target[shuffled_indices]

svm_classifier = SVC(C=1, gamma='auto', probability=True, random_state=246, kernel="linear")
random_forest_classifier = RandomForestClassifier(bootstrap=False, n_estimators=120, max_depth=30, random_state=9636, n_jobs=-1)
logistic_regression_classifier = LogisticRegression(solver="liblinear", random_state=12300)

voting_classifier = VotingClassifier(
    estimators=[('lr', logistic_regression_classifier), ('rf', random_forest_classifier), ('svc', svm_classifier)],
    voting='soft'
)

# 교차 검증 수행
cross_val_scores = cross_val_score(voting_classifier, shuffled_train_data, shuffled_train_target, cv=10, scoring='accuracy', n_jobs=-1)
print(f"Cross Validation Scores: {cross_val_scores}")
print(f"Cross Validation Mean is {np.mean(cross_val_scores)}")


Cross Validation Scores: [0.99184783 0.98777174 0.99591837 0.99047619 0.98639456 0.98367347
 0.98503401 0.99455782 0.99183673 0.99591837]
Cross Validation Mean is 0.9903429089026915


In [60]:
# 시드 설정하기
seed = 90068

# 데이터 섞기
np.random.seed(seed)
shuffled_indices = np.random.permutation(len(train_data))
shuffled_train_data = train_data.values[shuffled_indices]
shuffled_train_target = train_target[shuffled_indices]

# 모델 초기화
logistic_regression_classifier = LogisticRegression(solver="liblinear", random_state=12300)
random_forest_classifier = RandomForestClassifier(bootstrap=False, n_estimators=120, max_depth=30, random_state=9636, n_jobs=-1)
svm_classifier = SVC(C=1, gamma='auto', probability=True, random_state=246, kernel="linear")

voting_classifier = VotingClassifier(
    estimators=[('lr', logistic_regression_classifier), ('rf', random_forest_classifier), ('svc', svm_classifier)],
    voting='soft'
)

# 교차 검증 수행
cross_val_scores = cross_val_score(voting_classifier, shuffled_train_data, shuffled_train_target, cv=10, scoring='accuracy', n_jobs=-1)
print(f"Cross Validation Scores: {cross_val_scores}")
print(f"Cross Validation Mean is {np.mean(cross_val_scores)}")


Cross Validation Scores: [0.99320652 0.99456522 0.99727891 0.9877551  0.98367347 0.99047619
 0.98367347 0.99047619 0.99727891 0.98911565]
Cross Validation Mean is 0.9907499630286898


In [61]:
# 시드 설정하기
seed = 21401

# 데이터 섞기
np.random.seed(seed)
shuffled_indices = np.random.permutation(len(train_data))
shuffled_train_data = train_data.values[shuffled_indices]
shuffled_train_target = train_target[shuffled_indices]

# 모델 초기화
logistic_regression_classifier = LogisticRegression(solver="liblinear", random_state=12300)
random_forest_classifier = RandomForestClassifier(bootstrap=False, n_estimators=120, max_depth=30, random_state=9636, n_jobs=-1)
svm_classifier = SVC(C=1, gamma='auto', probability=True, random_state=246, kernel="linear")

voting_classifier = VotingClassifier(
    estimators=[('lr', logistic_regression_classifier), ('rf', random_forest_classifier), ('svc', svm_classifier)],
    voting='soft'
)

# 교차 검증 수행
cross_val_scores = cross_val_score(voting_classifier, shuffled_train_data, shuffled_train_target, cv=10, scoring='accuracy', n_jobs=-1)
print(f"Cross Validation Scores: {cross_val_scores}")
print(f"Cross Validation Mean is {np.mean(cross_val_scores)}")


Cross Validation Scores: [0.99048913 0.98505435 0.99183673 0.99183673 0.99455782 0.99183673
 0.99591837 0.99047619 0.99047619 0.99319728]
Cross Validation Mean is 0.9915679532682639


#### 최종 값:0.9915679532682639

# 시드 설정하기
seed = 21401

# 데이터 섞기
np.random.seed(seed)
shuffled_indices = np.random.permutation(len(train_data))
shuffled_train_data = train_data.values[shuffled_indices]
shuffled_train_target = train_target[shuffled_indices]

# 모델 초기화
logistic_regression_classifier = LogisticRegression(solver="liblinear", random_state=12300)
random_forest_classifier = RandomForestClassifier(bootstrap=False, n_estimators=120, max_depth=30, random_state=9636, n_jobs=-1)
svm_classifier = SVC(C=1, gamma='auto', probability=True, random_state=246, kernel="linear")

voting_classifier = VotingClassifier(
    estimators=[('lr', logistic_regression_classifier), ('rf', random_forest_classifier), ('svc', svm_classifier)],
    voting='soft'
)

# 교차 검증 수행
cross_val_scores = cross_val_score(voting_classifier, shuffled_train_data, shuffled_train_target, cv=10, scoring='accuracy', n_jobs=-1)
print(f"Cross Validation Scores: {cross_val_scores}")
print(f"Cross Validation Mean is {np.mean(cross_val_scores)}")
